### ISSUE TITLE
[D8] Add a phone field type

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1342]
Part of #378:

![d8-phone_number_field_type](https://cloud.githubusercontent.com/assets/2423362/11021058/b920494a-8689-11e5-9a8b-76da748b3dc1.png)

**Comment by quicksketch** [comment]
Thanks @klonos!

**Comment by klonos** [comment]
What would be a realistic milestone for this one?

**Comment by klonos** [comment]
...setting it to 1.6.0, but if anyone thinks we can achieve this for 1.5.0 (to compensate for not shipping references perhaps?), then feel free to change the milestone.

**Comment by jenlampton** [comment]
I don't think this is an 80% use-case, and think we should set our priorities on things that are. We have a contrib module already. How about we watch usage statistics on that, and if it ends up near the top we can add it to a specific milestone then?

**Comment by domaingood** [comment]
I think Use "Regular Expression Validation"
That,s All.

**Comment by indigoxela** [comment]
The html5 tel form item type is already in core. ( `<input type="tel"...`)

See: https://github.com/backdrop/backdrop/blob/e3ebbec7b5875b83ee4b0acf6e81026f51c4ff1b/core/modules/system/system.module#L293

We have no telephone field type in core, but there's a [contrib module](https://github.com/backdrop-contrib/telephone)

**Comment by klonos** [comment]
> How about we watch usage statistics on that, and if it ends up near the top we can add it to a specific milestone then?

https://backdropcms.org/project/usage/telephone has been steadily, but slowly increasing. It's currently at ~90, which considering the ~1300+ backdrop sites, that's very low. So we should indeed just leave it to contrib for now, and reevaluate if the usage increases.

The only thing that I'm concerned about is that it's in D8 core already; so this is a feature parity issue 🤔

**Comment by stpaultim** [comment]
@klonos - I don't disagree with your point about usage, but another way of looking at it is that the telephone module is #43 (out of over 500) on the list of modules and only 10 modules have more than 200 users.

https://backdropcms.org/project/usage
